libevmasm/SemanticInformation.cpp

```diff
@@ -213,20 +213,15 @@ bool SemanticInformation::sideEffectFree(Instruction _instruction)
 	// These are not really functional.
 	assertThrow(!isDupInstruction(_instruction) && !isSwapInstruction(_instruction), AssemblyException, "");
 
-	InstructionInfo info = instructionInfo(_instruction);
-	switch (_instruction)
-	{
-	// All the instructions that merely read memory are fine
-	// even though they are marked "sideEffects" in Instructions.cpp
-	case Instruction::KECCAK256:
-	case Instruction::MLOAD:
+	return !instructionInfo(_instruction).sideEffects;
+}
+
+bool SemanticInformation::sideEffectFreeUpToMSize(Instruction _instruction)
+{
+	if (_instruction == Instruction::KECCAK256 || _instruction == Instruction::MLOAD)
```

Comment:
Don't other instructions such as `mstore` also affect `msize()`

Reply:
Yes, but they also return false for `sideEffectFree` - do you propose to rather list them here explicitly?

Reply:
Not important, but I was reading the code as

```
if (_instruction == Instruction::KECCAK256 || _instruction == Instruction::MLOAD || sideEffectFree(_instruction))
  return true;
return false;
```

and thinking if the `if` statement could be optimized for an early return.

Reply:
What do you mean, "optimized for an early return"?

Reply:
returning even before stepping into another function (e.g., `sideEffectFree`) if possible to save a few processor cycles.

Reply:
It does exactly that for keccak256 and mload. What else do you propose? Also, though shalt not make assumptions about how the optimizer deals with code ;)